okhttp-testing-support/src/main/java/okhttp3/OkHttpClientTestRule.kt

```diff
@@ -86,15 +91,8 @@ class OkHttpClientTestRule : TestRule {
         }
       }
 
-      private fun acquireClient() {
-        prototype = prototypes.poll() ?: freshClient()
-      }
-
       private fun releaseClient() {
-        prototype?.let {
-          prototypes.push(it)
-          prototype = null
-        }
+        (testClient?.dispatcher as? ExecutorService)?.shutdown()
```

Comment:
Are you sure the safe cast of dispatcher to ExecutorService actually fires?

Reply:
It is definitely wrong. Needed to be this:

```
        testClient?.dispatcher?.executorService?.shutdown()
```

Reply:
Great catch!